### DISCUSSION TITLE
Option to prematurely end execution of retry and parallel

### DISCUSSION THREAD

**Discussion by SaidbekAbdiganiev** [radashi-org/radashi#256] [Ideas]
Currently, there is no way to forcibly terminate execution of `retry` and `parallel` functions. This presents a problem in scenarios where:

- A large number of parameters are provided to `parallel`, or
- A high retry count is set for `retry`

In such cases, we must wait for all iterations to complete, which can lead to unnecessarily long execution times.

The simplest solution that comes to mind is just to use AbortController, but I'm not sure if this is good enough since AbortController tends to be slow in Node (or at least I heard so). Would you like me to open a PR with an implementation of this feature?

**Comment by MarlonPassos-git** [comment]
thinking about it, this `AbortController` feature seems to be the most used way to solve these problems. About performance, just implement it and do the benchmarks to know. For me it seems interesting, I've had to use it a lot with [`p-map`](https://github.com/sindresorhus/p-map).

Feel free to open a PR, you can use the [`p-map` implementation as a reference](https://github.com/sindresorhus/p-map/blob/main/index.js#L39)

**Comment by MarlonPassos-git** [comment]
the points raised in this discussion were resolved in the PR https://github.com/radashi-org/radashi/pull/262
